### PR TITLE
Switching to new FinalExecutionOutcome. NEP#17

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Might need to change how I handle mixed enums. 

PR in nearcore: https://github.com/nearprotocol/nearcore/pull/1350
NEP#17: https://github.com/nearprotocol/NEPs/pull/17

### CODE REVIEW

File: test/account.test.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ test('delete account', async() => {
         await reloaded.state();
         fail('Getting state of deleted account was sucessful');
     } catch (_) {
-        // Expected to throw
+        // Expected to throwroviders.test.js
```
Comment (vgrichina) [inline test/account.test.js]: looks accidental

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ test:
   tags:
     - gcloud
   script:
-    - git clone --depth 1 https://github.com/nearprotocol/nearcore.git nearcore
+    - git clone --depth 1 --branch staging https://github.com/nearprotocol/nearcore.git nearcore
```
Comment (vgrichina) [inline .gitlab-ci.yml]: I think it's good idea to do both, similar to how @janedegtiareva did for `near-shell`:
https://github.com/nearprotocol/near-shell/pull/138

File: test/account.test.js (hunk 1 above)
Comment (evgenykuzyakov) [reply test/account.test.js]: thanks

File: .gitlab-ci.yml (hunk 2 above)
Comment (evgenykuzyakov) [reply .gitlab-ci.yml]: I think master is not going to work, since this staging change is breaking the results.

**Pull request merged**